File: backend/Tickets.Domain/Models/Especificaciones/EspecificacionesTelefonoIP.cs
using System.Text.Json.Serialization;

namespace Tickets.Domain.Models.Especificaciones
{
    /// <summary>
    /// Especificaciones técnicas para Teléfonos IP
    /// Se almacena en EspecificacionesJson
    /// </summary>
    public class EspecificacionesTelefonoIP
    {
        [JsonPropertyName("extension")]
        public string? Extension { get; set; }

        [JsonPropertyName("numeroLineas")]
        public int? NumeroLineas { get; set; }

        [JsonPropertyName("protocoloVoIP")]
        public string? ProtocoloVoIP { get; set; } // SIP, H.323, SCCP

        [JsonPropertyName("codecsAudio")]
        public string? CodecsAudio { get; set; } // G.711, G.729, Opus

        [JsonPropertyName("pantalla")]
        public bool? Pantalla { get; set; }

        [JsonPropertyName("pantallaTamanioPulgadas")]
        public decimal? PantallaTamanioPulgadas { get; set; }

        [JsonPropertyName("pantallaColor")]
        public bool? PantallaColor { get; set; }

        [JsonPropertyName("pantallaTactil")]
        public bool? PantallaTactil { get; set; }

        [JsonPropertyName("poe")]
        public bool? PoE { get; set; }

        [JsonPropertyName("wifi")]
        public bool? Wifi { get; set; }

        [JsonPropertyName("bluetooth")]
        public bool? Bluetooth { get; set; }

        [JsonPropertyName("manosLibres")]
        public bool? ManosLibres { get; set; }

        [JsonPropertyName("headset")]
        public bool? Headset { get; set; }

        [JsonPropertyName("teclasFuncion")]
        public int? TeclasFuncion { get; set; }

        [JsonPropertyName("moduloExpansion")]
        public bool? ModuloExpansion { get; set; }

        [JsonPropertyName("puertoUSB")]
        public bool? PuertoUSB { get; set; }

        [JsonPropertyName("grabacionLlamadas")]
        public bool? GrabacionLlamadas { get; set; }

        [JsonPropertyName("directorio")]
        public int? DirectorioContactos { get; set; } // Número de contactos

        [JsonPropertyName("versionFirmware")]
        public string? VersionFirmware { get; set; }

        [JsonPropertyName("servidorSIP")]
        public string? ServidorSIP { get; set; }
    }
}

File: frontend/lib/data/models/equipo/especificaciones/especificaciones_telefono_ip.dart
import 'dart:convert';

/// Especificaciones técnicas para Teléfonos IP
class EspecificacionesTelefonoIP {
  final String? extension;
  final int? numeroLineas;
  final String? protocoloVoIP; // SIP, H.323, SCCP
  final String? codecsAudio; // G.711, G.729, Opus

  final bool? pantalla;
  final double? pantallaTamanioPulgadas;
  final bool? pantallaColor;
  final bool? pantallaTactil;

  final bool? poe;
  final bool? wifi;
  final bool? bluetooth;
  final bool? manosLibres;
  final bool? headset;

  final int? teclasFuncion;
  final bool? moduloExpansion;
  final bool? puertoUSB;
  final bool? grabacionLlamadas;
  final int? directorioContactos; // Número de contactos

  final String? versionFirmware;
  final String? servidorSIP;

  const EspecificacionesTelefonoIP({
    this.extension,
    this.numeroLineas,
    this.protocoloVoIP,
    this.codecsAudio,
    this.pantalla,
    this.pantallaTamanioPulgadas,
    this.pantallaColor,
    this.pantallaTactil,
    this.poe,
    this.wifi,
    this.bluetooth,
    this.manosLibres,
    this.headset,
    this.teclasFuncion,
    this.moduloExpansion,
    this.puertoUSB,
    this.grabacionLlamadas,
    this.directorioContactos,
    this.versionFirmware,
    this.servidorSIP,
  });

  factory EspecificacionesTelefonoIP.fromJson(Map<String, dynamic> json) {
    return EspecificacionesTelefonoIP(
      extension: json['extension'],
      numeroLineas: json['numeroLineas'],
      protocoloVoIP: json['protocoloVoIP'],
      codecsAudio: json['codecsAudio'],
      pantalla: json['pantalla'],
      pantallaTamanioPulgadas: json['pantallaTamanioPulgadas']?.toDouble(),
      pantallaColor: json['pantallaColor'],
      pantallaTactil: json['pantallaTactil'],
      poe: json['poe'],
      wifi: json['wifi'],
      bluetooth: json['bluetooth'],
      manosLibres: json['manosLibres'],
      headset: json['headset'],
      teclasFuncion: json['teclasFuncion'],
      moduloExpansion: json['moduloExpansion'],
      puertoUSB: json['puertoUSB'],
      grabacionLlamadas: json['grabacionLlamadas'],
      directorioContactos: json['directorioContactos'],
      versionFirmware: json['versionFirmware'],
      servidorSIP: json['servidorSIP'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'extension': extension,
      'numeroLineas': numeroLineas,
      'protocoloVoIP': protocoloVoIP,
      'codecsAudio': codecsAudio,
      'pantalla': pantalla,
      'pantallaTamanioPulgadas': pantallaTamanioPulgadas,
      'pantallaColor': pantallaColor,
      'pantallaTactil': pantallaTactil,
      'poe': poe,
      'wifi': wifi,
      'bluetooth': bluetooth,
      'manosLibres': manosLibres,
      'headset': headset,
      'teclasFuncion': teclasFuncion,
      'moduloExpansion': moduloExpansion,
      'puertoUSB': puertoUSB,
      'grabacionLlamadas': grabacionLlamadas,
      'directorioContactos': directorioContactos,
      'versionFirmware': versionFirmware,
      'servidorSIP': servidorSIP,
    };
  }

  static EspecificacionesTelefonoIP? fromJsonString(String? jsonString) {
    if (jsonString == null || jsonString.isEmpty) return null;
    try {
      return EspecificacionesTelefonoIP.fromJson(json.decode(jsonString));
    } catch (e) {
      return null;
    }
  }

  String toJsonString() {
    return json.encode(toJson());
  }
}
